src/components/PostsList.tsx

```diff
@@ -1,54 +1,58 @@
+import cn from 'classnames';
 
-import classNames from 'classnames';
-import React from 'react';
+import { useAppDispatch, useAppSelector } from '../app/hooks';
+import { setPost } from '../features/post/postSlice';
 import { Post } from '../types/Post';
 
-type Props = {
-  posts: Post[];
-  selectedPostId?: number;
-  onPostSelected: (post: Post | null) => void;
-};
+export const PostsList: FC = () => {
+  const { posts, selectedPost } = useAppSelector(state => state.posts);
+
+  const dispatch = useAppDispatch();
 
-export const PostsList: React.FC<Props> = ({
-  posts,
-  selectedPostId = 0,
-  onPostSelected,
-}) => (
-  <div data-cy="PostsList">
-    <p className="title">Posts:</p>
+  const handleSelectPost = (post: Post) => {
+    if (selectedPost?.id === post.id) {
+      dispatch(setPost(null));
+    } else {
+      dispatch(setPost(post));
+    }
+  };
 
-    <table className="table is-fullwidth is-striped is-hoverable is-narrow">
-      <thead>
-        <tr className="has-background-link-light">
-          <th>#</th>
-          <th>Title</th>
-          <th> </th>
-        </tr>
-      </thead>
+  return (
+    <div data-cy="PostsList">
+      <p className="title">Posts:</p>
 
-      <tbody>
-        {posts.map(post => (
-          <tr key={post.id} data-cy="Post">
-            <td data-cy="PostId">{post.id}</td>
-            <td data-cy="PostTitle">{post.title}</td>
-            <td className="has-text-right is-vcentered">
-              <button
-                type="button"
-                data-cy="PostButton"
-                className={classNames('button', 'is-link', {
-                  'is-light': post.id !== selectedPostId,
-                })}
-                onClick={() => {
-                  onPostSelected(post.id === selectedPostId ? null : post);
-                }}
-              >
-                {post.id === selectedPostId ? 'Close' : 'Open'}
-              </button>
-            </td>
+      <table className="table is-fullwidth is-striped is-hoverable is-narrow">
+        <thead>
+          <tr className="has-background-link-light">
+            <th>#</th>
+            <th>Title</th>
+            <th> </th>
           </tr>
-        ))}
-      </tbody>
-    </table>
-  </div>
-);
+        </thead>
+
+        <tbody>
+          {posts.map(post => (
+            <tr key={post.id} data-cy="Post">
+              <td data-cy="PostId">{post.id}</td>
+              <td data-cy="PostTitle">{post.title}</td>
+              <td className="has-text-right is-vcentered">
+                <button
+                  type="button"
+                  data-cy="PostButton"
+                  className={cn('button', 'is-link', {
+                    'is-light': post.id !== selectedPost?.id,
+                  })}
+                  onClick={() => handleSelectPost(post)}
+                >
+                  {post.id === selectedPost?.id ? 'Close' : 'Open'}
+                </button>
+              </td>
+            </tr>
```

Comment:
Consider creating a separate component for the post and using destruction assignment for the post object